src/components/Search/Lookup.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, Props } from 'react';
+import styled from 'styled-components';
+// import Icon from './FontIcons.tsx';
+import { backgroundColor, textColor, borderColor } from '../SETThemeProvider';
+
+const states = (states) => {
+    if (!Array.isArray(states)) {
+        states = [states];
+    }
+    return (props) => {
+        return states.map((state) => props[state] ? '&&' : `&:${state}`).join(',');
+    }
+}
+
+type IconName = string;
+
+interface LookupProps {
+    placeholder: string;
+    buttonText: string;
+    iconName: IconName;
+    onRequestSearch(value: string): string;
+}
+
+export default function Lookup({ placeholder, buttonText, iconName, onRequestSearch }: LookupProps) {
+
+    function handleRequestSearch(value: string, onRequestSearch: (value: string) => string) {
```

Comment:
If handleRequestSearch is inside Lookup function then you don't need to provide onRequestSearch as a param, instead you can just use it

```suggestion
    function handleRequestSearch(value: string) {
```